github/index.js

```diff
@@ -53,23 +53,52 @@ class GitHub {
     const { installationId, name } = organization
     const token = await this.app.getInstallationAccessToken({ installationId })
 
-    // get org admins
-    const admins = await this.got.paginate.all(
-      `https://api.github.com/orgs/${name}/members`, {
-        searchParams: {
-          role: 'admin',
-          per_page: 100
-        },
-        responseType: 'json',
-        headers: {
-          accept: 'application/vnd.github.v3+json',
-          Authorization: `Bearer ${token}`
+    // first determine the user's current username from their ID using an undocumented endpoint
+    // if this fails, we fall back to accessing the list of public members
+    try {
+      const { body: user } = await this.got.get(`https://api.github.com/user/${userGitHubId}`)
+      const { login: username } = user
+
+      try {
+        const { body: membership } = await this.got.get(`https://api.github.com/orgs/${name}/memberships/${username}`, {
+          responseType: 'json',
+          headers: {
+            accept: 'application/vnd.github.v3+json',
+            Authorization: `Bearer ${token}`
+          }
+        })
+
+        const { role, state } = membership
+        return role === 'admin' && state === 'active'
+      } catch (e) {
+        const { response } = e
+        if (response && response.statusCode === 404) {
```

Comment:
is this assuming that they've also updated the flossbank permissions to allow org membership access?

if so: i'd feel better if we threw here as well so that we go into the catch and at least try to see if they're a public member

Reply:
no, 404 means that the user is not in the organization

Reply:
we'd get a 403 if they hadn't updated the permissions, in which case it would throw